### PR TITLE
Feature/search bar

### PR DESCRIPTION
**Pull request by phongduong-dp**
# Description & Technical Solution

Create search bar component used for header on PC layout.
Component will work on Enter or by clicking search icon button.
Component is not responsive since another component for mobile will be created.
I tried to remove browser default behavior but it does not work.

# Checklist

- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] Already rebased against main branch.

# Screenshots

***PC layout***

https://github.com/its-rav/namvietstore-fe/assets/92157056/aec9fb27-d030-4ff9-8205-ac7d185a61ed

### CODE REVIEW

File: packages/core-components/src/components/SearchBar.tsx [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+};
+
+const SearchBar: React.FC<SearchBarProps> = ({
+  handleSubmit,
+  searchIcon,
+  placeholders,
+}) => {
+  return (
+    <form
+      onSubmit={handleSubmit}
```
Comment (its-rav) [inline packages/core-components/src/components/SearchBar.tsx]: How do we get the query text inside this handler

File: packages/core-components/src/components/SearchBar.tsx [hunk 2]
```diff
@@ -0,0 +1,31 @@
+import React from 'react';
+
+type SearchBarProps = {
+  handleSubmit: () => void;
+  searchIcon: React.ReactNode;
```
Comment (its-rav) [inline packages/core-components/src/components/SearchBar.tsx]: do we really need this in the prop, as it is already a 'search icon'!?

File: playground/src/FeaturesCard.stories.tsx [hunk 3]
```diff
@@ -31,20 +31,6 @@ export const Default: Story = () => (
       justifyContent: "center",
     }}
   >
-    <FeaturesCard featuresCardItem={featuresCardItem} />{" "}
```
Comment (its-rav) [inline playground/src/FeaturesCard.stories.tsx]: This change is not related to the feature, normally we should have this at another PR but we can let it slide this time

File: packages/core-components/src/components/SearchBar.tsx [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+  placeholders: string;
+};
+
+const SearchBar: React.FC<SearchBarProps> = ({
+  handleSubmit,
+  searchIcon,
+  placeholders,
```
Comment (its-rav) [inline packages/core-components/src/components/SearchBar.tsx]: placeholder you mean?

**Review by its-rav — CHANGES_REQUESTED** [review]

File: packages/core-components/src/components/SearchBar.tsx (hunk 4 above)
Comment (phongduong-dp) [reply packages/core-components/src/components/SearchBar.tsx]: Placeholders is the string shows default inside the input box (the "Search all products" part).

I think we will have VN and EN version for this, so I put it as a prop.


search

Comment (its-rav) [reply packages/core-components/src/components/SearchBar.tsx]: the plural naming I mean

File: packages/core-components/src/components/SearchBar.tsx (hunk 2 above)
Comment (its-rav) [reply packages/core-components/src/components/SearchBar.tsx]: pump this up

Comment (its-rav) [reply packages/core-components/src/components/SearchBar.tsx]: nvm

File: packages/core-components/src/components/SearchBar.tsx (hunk 1 above)
Comment (its-rav) [reply packages/core-components/src/components/SearchBar.tsx]: normally we have name  attr in input tag, and when we submit the form, the field will automately submit with the corresponding field
In this case, I don't think we need form personally. We can expose out onSubmit with query text as the param and call it on button click, and even better, expose onQueryChange for whenever the input value change.

Comment (its-rav) [reply packages/core-components/src/components/SearchBar.tsx]: FYI, we can use `useRef`/`createRef` to get the value of the input at the time

**Comment by its-rav** [comment]
conflict

**Comment by phongduong-dp** [comment]
> conflict

solved

**Review by its-rav — APPROVED** [review]

**Pull request merged**